src/Tests/Analyzers.Tests/RCS1216UnnecessaryUnsafeContextTests.cs

```diff
@@ -0,0 +1,249 @@
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Roslynator.CSharp.CodeFixes;
+using Roslynator.Testing.CSharp;
+using Xunit;
+
+namespace Roslynator.CSharp.Analysis.Tests;
+
+public class RCS1216UnnecessaryUnsafeContextTests : AbstractCSharpDiagnosticVerifier<UnnecessaryUnsafeContextAnalyzer, UnnecessaryUnsafeContextCodeFixProvider>
+{
+    public override DiagnosticDescriptor Descriptor { get; } = DiagnosticRules.UnnecessaryUnsafeContext;
+
+    [Trait(Traits.Analyzer, DiagnosticIdentifiers.UnnecessaryUnsafeContext)]
+    [Theory]
+    [InlineData(@"class")]
+    [InlineData(@"interface")]
+    [InlineData(@"record")]
+    [InlineData(@"struct")]
+    [InlineData(@"record struct")]
+    public async Task Test_Type(string typeKeyword)
+    {
+        var source = @"
+    unsafe <type_keyword> C
+    {
+        void M()
+        {
+            [|unsafe|] {
+                var x = 1;
+            }
+        }
+    }
+".Replace("<type_keyword>", typeKeyword);
+        var expected = @"
+    unsafe <type_keyword> C
+    {
+        void M()
+        {
+            {
+                var x = 1;
+            }
+        }
+    }
+".Replace("<type_keyword>", typeKeyword);
+        await VerifyDiagnosticAndFixAsync(source, expected, options: Options.WithAllowUnsafe(true));
+    }
+
+    [Theory]
+    [InlineData(@"void M()")]
+    [InlineData(@"C()")]
+    [InlineData(@"static void M()")]
+    public async Task Test_Member(string memberSignature)
+    {
+        var source = @"
+    class C
+    {
+        unsafe <member_signature>
+        {
+            [|unsafe|] {
+                var x = 1;
+            }
+        }
+    }
+".Replace("<member_signature>", memberSignature);
```

Comment:
This "replace" pattern is not used anywhere in the unit tests. Please create separate method for each case.